zsl/resource/model_resource.py

```diff
@@ -169,11 +168,19 @@ def __init__(self, model_cls=None):
         mapper = class_mapper(self.model_cls)
         self._model_pk = mapper.primary_key[0]
         self._model_columns = [column.key for column in mapper.column_attrs]
+        self._related_columns = [column.key for column in mapper.relationships]
+
+    def to_filter(self, query, arg):
+        return filter_from_url_arg(self.model_cls, query, arg)
+
+    def add_related(self, query, related):
+        return apply_related(self.model_cls, query, related)
+
+    def set_ordering(self, query, arg):
+        return order_from_url_arg(self.model_cls,  query, arg)
```

Comment:
I moved them from the constructor, there's no reason to create them in the constructor.